app/parse.py

```diff
@@ -15,8 +23,105 @@ class Product:
     num_of_reviews: int
 
 
+class Driver:
+    _instance = None
+
+    def __new__(cls, *args, **kwargs) -> "Driver":
+        if cls._instance is None:
+            cls._instance = super().__new__(cls, *args, **kwargs)
+        return cls._instance
+
+    def __init__(self) -> None:
+        options = Options()
+        options.add_argument("--headless")
+        options.add_argument("--disable-gpu")
+        self.driver = webdriver.Chrome(options=options)
+
+    def get_driver(self) -> WebDriver:
+        return self.driver
+
+
+_driver = Driver().get_driver()
+
+pages = [
+    {"name": "home", "url": BASE_URL},
+    {"name": "computers", "url": urljoin(BASE_URL, "computers")},
+    {"name": "laptops", "url": urljoin(BASE_URL, "computers/laptops")},
+    {"name": "tablets", "url": urljoin(BASE_URL, "computers/tablets")},
+    {"name": "phones", "url": urljoin(BASE_URL, "phones")},
+    {"name": "touch", "url": urljoin(BASE_URL, "phones/touch")}
+]
+
+
+def parse_single_product(product_card: WebElement) -> Product:
+    title = product_card.find_element(By.CLASS_NAME,
+                                      "title").get_attribute("title")
+    description = product_card.find_element(By.CLASS_NAME, "description").text
+    price = float(product_card.find_element(By.CLASS_NAME, "price").text[1:])
+    stars = product_card.find_elements(By.CLASS_NAME, "ws-icon-star")
+    rating = len(stars)
+    reviews_string = product_card.find_element(By.CLASS_NAME, "ratings").text
+    num_of_reviews = int(reviews_string.split(" ")[0])
+
+    return Product(
+        title=title,
+        description=description,
+        price=price,
+        rating=rating,
+        num_of_reviews=num_of_reviews
+    )
+
+
+def parse_page(url: str) -> list[Product]:
+    _driver.get(url)
+
+    try:
+        WebDriverWait(_driver, 5).until(
+            ec.element_to_be_clickable((By.CLASS_NAME, "acceptCookies"))
+        ).click()
+    except (TimeoutException, ElementNotInteractableException):
+        pass
```

Comment:
The exception handling here is quite broad. Consider logging the exceptions or providing more specific handling to understand why the element might not be interactable or why a timeout occurs. This will help in debugging and improving the robustness of the script.